src/encoded/types/patient.py

```diff
@@ -466,51 +475,62 @@ def surgery_summary(self, request, surgery=None):
-                for procedure_record in surgery_procedure:
-                    procedure_obj = request.embed(procedure_record, '@@object')
-                    if procedure_obj['procedure_type'] == "Nephrectomy":
+                surgery_path_report = surgery_object['pathology_report']
+                for path_report in surgery_path_report:
+                    path_report_obj = request.embed(path_report, '@@object')
+                    if path_report_obj['path_source_procedure'] == "path_nephrectomy":
                         nephrectomy_dates.append(surgery_object['date'])
-                    elif  procedure_obj['procedure_type'] == "Biopsy" or procedure_obj['procedure_type'] == "Metastectomy":
+                    elif  path_report_obj['path_source_procedure'] == "path_biopsy" or path_report_obj['path_source_procedure'] == "path_metasis":
                         non_nephrectomy_dates.append(surgery_object['date'])
+
         if len(nephrectomy_dates) > 0 :
-            nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
-            diagnosis_date = nephrectomy_dates[0]
-        else:
-            if len(radiation) > 0:
-                # add radiation dates
-                for radiation_record in radiation:
-                    radiation_object = request.embed(radiation_record, '@@object')
-                    non_nephrectomy_dates.append(radiation_object['start_date'])
-            if len(medication) > 0:
-                # add medication dates
-                for medication_record in medication:
-                    medication_object = request.embed(medication_record, '@@object')
-                    non_nephrectomy_dates.append(medication_object['start_date'])
-
-            if len(non_nephrectomy_dates) > 0:
+                nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+                diagnosis_date = nephrectomy_dates[0]
+                surgery_dates.append(nephrectomy_dates[0])
+                diagnosis_source = "Pathology Report"
+
+        elif len(non_nephrectomy_dates) > 0:
+            non_nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+            surgery_dates.append(non_nephrectomy_dates[0])
+            surgery_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
+            diagnosis_date = surgery_dates[0]
+            diagnosis_source = "Pathology Report"
+
+        elif diagnosis_date_tumor_registry is not None:
+             diagnosis_date = diagnosis_date_tumor_registry
+        elif len(medication) > 0:
+            diagnosis_source = "Medication"
+            for medication_record in medication:
+                medication_object = request.embed(medication_record, '@@object')
+                non_nephrectomy_dates.append(medication_object['start_date'])
                 non_nephrectomy_dates.sort(key = lambda date: datetime.strptime(date, '%Y-%m-%d'))
```

Comment:
sort function should happen outside for loop, after all the medication dates are added into list "non_nephrectomy_dates".

Reply:
Fixed this